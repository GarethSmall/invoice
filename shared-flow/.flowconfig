[ignore]
[include]

[libs]

[options]
module.system=haste
esproposal.decorators=ignore
esproposal.class_static_fields=enable
esproposal.class_instance_fields=enable

munge_underscores=true

suppress_type=$FlowIssue
suppress_type=$FlowFixMe
suppress_type=$FixMe

suppress_comment=\\(.\\|\n\\)*\\$FlowFixMe\\($\\|[^(]\\|(\\(>=0\\.\\(2[0-3]\\|1[0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)
suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(>=0\\.\\(2[0-3]\\|1[0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)?:? #[0-9]+
suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy

module.file_ext=.js
module.file_ext=.json
module.file_ext=.jsx

[version]
0.67.1
