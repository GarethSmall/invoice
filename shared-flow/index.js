// @flow
export * from './src/Product';
export * from './src/Order';
export * from './src/OrderItem';