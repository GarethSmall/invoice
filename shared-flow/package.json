{
  "name": "shared-flow",
  "version": "1.0.0",
  "description": "Flow typing for our invoice",
  "main": "src/index.js",
  "scripts": {
    "flow": "flow check",
    "lint": "eslint --ext .js,.jsx ./src",
    "lint-fix": "eslint --ext js,jsx ./src --fix",
    "prettier": "prettier-eslint \"./src/**/*.{scss,js}\" --write"
  },
  "author": "Gareth Small",
  "license": "ISC",
  "devDependencies": {
    "babel": "^6.23.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.2",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-decorators": "^6.24.1",
    "babel-plugin-transform-decorators-legacy": "^1.3.4",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-flow": "^6.23.0",
    "eslint": "^4.12.0",
    "eslint-config-prettier": "^2.9.0",
    "eslint-loader": "^1.9.0",
    "eslint-plugin-flow": "^2.29.1",
    "eslint-plugin-flowtype": "^2.46.1",
    "eslint-plugin-import": "^2.8.0",
    "eslint-plugin-jsx-a11y": "^6.0.2",
    "eslint-plugin-prettier": "^2.6.0",
    "flow-bin": "^0.67.1",
    "prettier-eslint-cli": "^4.7.1"
  }
}
