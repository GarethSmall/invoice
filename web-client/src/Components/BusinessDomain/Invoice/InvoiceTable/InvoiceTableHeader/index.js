//@flow
import InvoiceTableHeader from './InvoiceTableHeader';
export default InvoiceTableHeader;
