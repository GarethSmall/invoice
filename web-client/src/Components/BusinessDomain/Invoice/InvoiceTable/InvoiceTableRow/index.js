//@flow
import InvoiceTableRow from './InvoiceTableRow';
export default InvoiceTableRow;
