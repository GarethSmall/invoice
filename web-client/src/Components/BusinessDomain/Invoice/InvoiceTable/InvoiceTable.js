// @flow
import React, { PureComponent } from 'react';
import { Order, Product } from 'shared-flow';
import InvoiceTableHeader from './InvoiceTableHeader';
import InvoiceTableRow from './InvoiceTableRow';

type Props = {
  products : Product[],
  order : Order,
};

export default class InvoiceTable extends PureComponent<Props> {
  render() {
    const { order } = this.props;
    return (
      <table>
        <InvoiceTableHeader />
        <tbody>
          {order.items.map(item => <InvoiceTableRow key={`${item._id}`} item={item} />)}
        </tbody>
      </table>
    );
  }
}