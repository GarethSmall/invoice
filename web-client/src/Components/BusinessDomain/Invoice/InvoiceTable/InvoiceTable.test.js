// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import shallow from 'enzyme/shallow';
import InvoiceTable from './';
import { Order } from 'shared-flow';

const { describe, test, expect } = global;

Enzyme.configure({ adapter: new Adapter() });

describe('InvoiceTable', () => {
  test('should render', () => {
    const mockOrder : Order = {
      _id : '1',
      items : [
        {
          _id: '1',
          order: '1',
          product: {
            _id: '1',
            name: 'Test T-shirt',
            price: 32.00,
          },
          quantity: 2,
          price: 64.00,
        },
        {
          _id: '2',
          order: '1',
          product: {
            _id: '2',
            name: 'Test Jeans',
            price: 75.00,
          },
          quantity: 2,
          price: 150.00,
        },
      ],
      subTotalPrice: 214.00,
      taxPrice: 10.70,
      totalPrice: 224.70,
    };
    const component = shallow((
      <InvoiceTable order={mockOrder} />
    ));
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
