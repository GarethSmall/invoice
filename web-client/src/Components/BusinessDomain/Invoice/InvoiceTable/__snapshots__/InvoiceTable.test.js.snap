// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceTable should render 1`] = `
<table>
  <InvoiceTableHeader />
  <tbody>
    <InvoiceTableRow
      item={
        Object {
          "_id": "1",
          "order": "1",
          "price": 64,
          "product": Object {
            "_id": "1",
            "name": "Test T-shirt",
            "price": 32,
          },
          "quantity": 2,
        }
      }
      key="1"
    />
    <InvoiceTableRow
      item={
        Object {
          "_id": "2",
          "order": "1",
          "price": 150,
          "product": Object {
            "_id": "2",
            "name": "Test Jeans",
            "price": 75,
          },
          "quantity": 2,
        }
      }
      key="2"
    />
  </tbody>
</table>
`;
